File: code/318/solution.cpp
#include <iostream>/*{{{*/
#include <math.h>
using namespace std; /*}}}*/
int main() {
    long long ans = 0;
    for (int p = 1; p < 2011; p++) {
        for (int q = p + 1; p + q <= 2011; q++) {
            double buf = p + q - 2 * sqrt(p * q);
            if (buf <= 0 || buf >= 1)
                continue;
            ans += ceil(-2011 / log10(buf));
        }
    }
    cout << ans << endl;
    return 0;
}

File: code/318/solution.go
package main /*{{{*/

import (
	"fmt"
	"math"
)

func main() {
	ans := int64(0)
	for p := 1; p < 2011; p++ {
		for q := p + 1; p+q <= 2011; q++ {
			buf := float64(p+q) - 2*math.Sqrt(float64(p*q))
			if 0 < buf && buf < 1 {
				ans += int64(math.Ceil(-2011 / math.Log10(buf)))
			}
		}
	}
	fmt.Println(ans)
}
